App/src/main/java/cc/softwarefactory/lokki/android/MainActivity.java

```diff
@@ -277,22 +280,15 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
     private void toggleVisibility() {
 
-        MainApplication.visible = !MainApplication.visible;
-        int visibility_mode = MainApplication.visible ? 0 : 1;
-        PreferenceUtils.setValue(this, PreferenceUtils.KEY_SETTING_VISIBILITY, String.valueOf(visibility_mode));
+        Utils.setVisibility(!MainApplication.visible, MainActivity.this); // oh the side effects...
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
```

Comment:
Here you should get the shared prefs with the application contexts as you are getting the shared prefs in `MainApplication.java` with the application context.